src/braket/default_simulator/openqasm/_helpers/casting.py

```diff
@@ -99,8 +113,9 @@ def _(into: UintType, variable: LiteralType) -> IntegerLiteral:
     return IntegerLiteral(value)
 
 
-@cast_to.register
-def _(into: FloatType, variable: LiteralType) -> FloatLiteral:
+@cast_to.register(FloatType)
+@cast_to.register(AngleType)
+def _(into, variable: LiteralType) -> FloatLiteral:
```

Comment:
The no size case is shared between float and angle, but for specified-bit variables, the behavior is different

Reply:
I think the way to handle this would be to store fixed-sized angles as `BitstringLiteral`s and update `handle_parameter_value` to do the % 2pi math when building the circuit